### DISCUSSION TITLE
how markerts should behave when facing double inheritance.

### DISCUSSION THREAD

**Discussion by NicolasTriquenaux** [pytest-dev/pytest#8400] [Q&A] [answered]
Dear Community,

I am acknoledging the current behavior and i would like to verify that it is what should be expected in the current situation.

Let's have some example : 
- Case 1
```
class A (object):
   pass

@pytest.mark.mymarker("some other value")
class B (object):
   pass

class C (A, B):
   pass
```
When printing the marker list applied to class C i get : `@pytest.mark.mymarker("some other value")`, which seems logical to me

- Case 2
```
@pytest.mark.othermarker("some value")
class A (object):
   pass

@pytest.mark.mymarker("some other value")
class B (object):
   pass

class C (A, B):
   pass
```
When printing the marker list applied to class C i get : `@pytest.mark.othermarker("some value")`, and not the fusion of both markers. Is this behavior is expected ? i failed to find in pytest sources where the markers inheritance where handled.

Regards

**Comment by RonnyPfannschmidt** [comment, marked as answer]
marker extraction currently does not handle inheritance adequately i believe, i'll crosscheck

*edit*

a temporary workaround is to copy together the pytestmark attributes of both base classes to get all marks on the class,
if you choose that solution please link the discussion or the issue i created so its clear its a workaround for a bug/oversight

**Reply by RonnyPfannschmidt** [reply]
indeed, this is a oversight - https://github.com/pytest-dev/pytest/blob/19a2f7425ddec3b614da7c915e0cf8bb24b6906f/src/_pytest/mark/structures.py#L371-L376 does not handle the mro at all

so inheritance currently does not correctly integrate markers

**Reply by RonnyPfannschmidt** [reply]
i created #8401

**Reply by NicolasTriquenaux** [reply]
@RonnyPfannschmidt : i do thank you for your swift answer.
